homeassistant/components/simplisafe/__init__.py

```diff
@@ -200,3 +184,33 @@ def _async_save_refresh_token(hass, config_entry, token):
     remove_listener()
 
     return True
+
+
+class SimpliSafe:
+    """Define a SimpliSafe API object."""
+
+    def __init__(self, hass, config_entry, systems):
+        """Initialize."""
+        self._config_entry = config_entry
+        self._hass = hass
+        self.systems = systems
+
+    async def async_update(self):
+        """Get updated data from SimpliSafe."""
+        tasks = [system.update() for system in self.systems.values()]
+
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+        for system, result in zip(self.systems.values(), results):
+            if isinstance(result, SimplipyError):
+                _LOGGER.error(
+                    'There was error updating "%s": %s', system.address,
+                    result)
+                continue
+
+            _LOGGER.debug('Updated status of "%s"', system.address)
+            async_dispatcher_send(
```

Comment:
Sequence wise, would it make more sense to do this last -- after saving the refresh token if necessary?

Reply:
Doesn't really matter: saving the refresh token doesn't affect the downstream components that rely on the update dispatch. That said, I hear what you're saying: logically, it makes sense that dispatch is the last thing that happens. 👍